Errors in `package.xml` doesn't stop action, failed dependency fullfilment revealed later at build

## Description
```
  + rosdep install -r --from-paths src/iodztm29wur/ros2_dummy_tools --ignore-src --skip-keys 'rti-connext-dds-5.3.1 rti-connext-dds-6.0.1 ' --rosdistro iron -y
  Error(s) in package '/__w/ros2_dummy_tools/ros2_dummy_tools/ros_ws/src/iodztm29wur/ros2_dummy_tools/./package.xml':
  Error(s):
  - The generic dependency on 'rclcpp_components' is redundant with: build_depend
  + true
```
This failure in invocation of `bash -c,./install_rosdeps.sh iron` continues w/o stopping.

Later CMake fails as cannot find config files for dependencies required.

### Expected Behavior
**Action shall stop on incorrect `package.xml` and let CI report this as error.**

### Actual Behavior
**As shown above, it carries on despite mistake in `packages.xml` then fails CMake with cryptic:**
```
 CMake Error at CMakeLists.txt:66 (find_package):
    By not providing "Findnaporet_foobar_msgs.cmake" in CMAKE_MODULE_PATH this
    project has asked CMake to find a package configuration file provided by
    "naporet_foobar_msgs", but CMake did not find one.
  
    Could not find a package configuration file provided by
    "naporet_foobar_msgs" with any of the following names:
  
      naporet_foobar_msgsConfig.cmake
      naporet_foobar_msgs-config.cmake
  
    Add the installation prefix of "naporet_foobar_msgs" to CMAKE_PREFIX_PATH
    or set "naporet_foobar_msgs_DIR" to a directory containing one of the above
    files.  If "naporet_foobar_msgs" provides a separate development package or
    SDK, be sure it has been installed.
  
  
  -- Configuring incomplete, errors occurred!
  See also "/__w/ros2_dummy_tools/ros2_dummy_tools/ros_ws/build/naporet_dummy_tools/CMakeFiles/CMakeOutput.log".
  ---
  --- stderr: naporet_dummy_tools
  CMake Error at CMakeLists.txt:66 (find_package):
    By not providing "Findnaporet_foobar_msgs.cmake" in CMAKE_MODULE_PATH this
    project has asked CMake to find a package configuration file provided by
    "naporet_foobar_msgs", but CMake did not find one.
  
    Could not find a package configuration file provided by
    "naporet_foobar_msgs" with any of the following names:
  
      naporet_foobar_msgsConfig.cmake
      naporet_foobar_msgs-config.cmake
  
    Add the installation prefix of "naporet_foobar_msgs" to CMAKE_PREFIX_PATH
    or set "naporet_foobbar_msgs_DIR" to a directory containing one of the above
    files.  If "naporet_foobar_msgs" provides a separate development package or
    SDK, be sure it has been installed.
  
  
  ---
  Failed   <<< naporet_dummy_tools [3.91s, exited with code 1]
```

## To Reproduce
1. Leave repeated dependency in `packages.xml`
2. Run CI

## System (please complete the following information)
 - OS: Ubuntu Jammy, Ubuntu Noble
 - ROS 2 Distro: Humble, Iron, Jazzy, Rolling

** For ROS 2 bugs - please attach a VCS.repos or Dockerfile if possible to help us reproduce the environment **

## Additional context
We're heavy user of your actions, we rely on their trustworthiness. This problem fails the trust and takes time of inexperienced engineers to figure out what they did wrong.

--- a/src/action-ros-ci.ts
+++ b/src/action-ros-ci.ts
@@ -202,7 +202,7 @@ async function installRosdeps(
 	# see https://index.ros.org/doc/ros2/Installation/Foxy/Linux-Development-Setup/#install-dependencies-using-rosdep
 	rosdep install -r --from-paths $package_paths --ignore-src --skip-keys "rti-connext-dds-5.3.1 rti-connext-dds-6.0.1 ${filterNonEmptyJoin(
 		skipKeys,
-	)}" --rosdistro $DISTRO -y || true`;
+	)}" --rosdistro $DISTRO -y`;
 	fs.writeFileSync(scriptPath, scriptContent, { mode: 0o766 });
 
 	let exitCode = 0;

